lib/puma/binder.rb

```diff
@@ -246,9 +240,11 @@ def parse(binds, logger, log_msg = 'Listening')
             ios_len = @ios.length
             io = add_ssl_listener uri.host, uri.port, ctx
 
+            uri_query = params.map { |k, v| "#{k}=#{v}" }.join('&')
+
             @ios[ios_len..-1].each do |i|
               addr = loc_addr_str i
-              logger.log "* #{log_msg} on ssl://#{addr}?#{uri.query}"
+              logger.log "* #{log_msg} on ssl://#{addr}?#{uri_query}"
```

Comment:
Why change this? Looks to me like `uri.query` is still available?

Reply:
So `uri.query` as `uri` comes from https://github.com/puma/puma/blob/master/lib/puma/dsl.rb#L66-L67 (where there is no possibility to detect localhost gem, that was in my try in the first attempt) so it gets the URI with empty `key` and `cert` values (`&key=&cert=`).
These empty values are transformed to Hash and stored in `params` variable here https://github.com/puma/puma/blob/master/lib/puma/binder.rb#L230, and then the current implementation is transforming them (default localhost gem paths). In the end only `params` is "enriched" by new values, thus I'm transforming it back from Hash to String